libbeat/docs/http-endpoint.asciidoc

```diff
@@ -0,0 +1,149 @@
+//// Use the following include to pull this content into a doc file:
+//// include::../../libbeat/docs/http-endpoint.asciidoc[]
+//////////////////////////////////////////////////////////////////////////
+
+[[http-endpoint]]
+== HTTP Endpoint
+
+experimental[]
+
+{beatname_uc} can expose internal metrics through a HTTP endpoint. For security
+reasons the endpoint is disabled by default.
+
+The HTTP endpoint has the following configuration settings:
+
+`http.enabled`:: (Optional) Enable the http endpoint. Default is `false`.
+`http.host`:: (Optional) Bind to this hostname or IP address.
+It is recommended to use only localhost. Default is `localhost`
+`http.port`:: (Optional) Port on which the HTTP endpoint will bind. Default is `5066`.
+
+[float]
+=== Paths
+
+This is the list of paths you can access. For pretty JSON output append ?pretty to the URL.
+
+
+`/` provides basic info from the {beatname_uc}. Example:
+
+[source,json]
+----
+{
+  "beat": "metricbeat",
+  "hostname": "example.lan",
+  "name": "example.lan",
+  "uuid": "34f6c6e1-45a8-4b12-9125-11b3e6e89866",
+  "version": "6.4.0"
+}
+----
+
+`/stats` reports internal metrics. Example:
+
+[source,json]
+----
+{
+  "beat": {
+    "cpu": {
+      "system": {
+        "ticks": 1710,
+        "time": {
+          "ms": 1712
+        }
+      },
+      "total": {
+        "ticks": 3420,
+        "time": {
+          "ms": 3424
+        },
+        "value": 3420
+      },
+      "user": {
+        "ticks": 1710,
+        "time": {
+          "ms": 1712
+        }
+      }
+    },
+    "info": {
+      "ephemeral_id": "ab4287c4-d907-4d9d-b074-d8c3cec4a577",
+      "uptime": {
+        "ms": 195547
+      }
+    },
+    "memstats": {
+      "gc_next": 17855152,
+      "memory_alloc": 9433384,
+      "memory_total": 492478864,
+      "rss": 50405376
+    }
+  },
+  "libbeat": {
+    "config": {
+      "module": {
+        "running": 0,
+        "starts": 0,
+        "stops": 0
+      },
+      "reloads": 1
+    },
+    "output": {
+      "events": {
+        "acked": 0,
+        "active": 0,
+        "batches": 0,
+        "dropped": 0,
+        "duplicates": 0,
+        "failed": 0,
+        "total": 0
+      },
+      "read": {
+        "bytes": 0,
+        "errors": 0
+      },
+      "type": "elasticsearch",
+      "write": {
+        "bytes": 0,
+        "errors": 0
+      }
+    },
+    "pipeline": {
+      "clients": 6,
+      "events": {
+        "active": 716,
+        "dropped": 0,
+        "failed": 0,
+        "filtered": 0,
+        "published": 716,
+        "retry": 278,
+        "total": 716
+      },
+      "queue": {
+        "acked": 0
+      }
+    }
+  },
+  "system": {
+    "cpu": {
+      "cores": 4
+    },
+    "load": {
+      "1": 2.22,
+      "15": 1.8,
+      "5": 1.74,
+      "norm": {
+        "1": 0.555,
+        "15": 0.45,
+        "5": 0.435
+      }
+    }
+  }
+}
+----
+
+The actual output may contain more metrics specific to {beatname_uc}
```

Comment:
Would it make sense to add an example for each beat? Would be fairly easy to add conditional coding around the examples. However, might be a bit of work to maintain the examples over time. If the example that you show here makes sense for all the Beats, then a single generic example will definitely be easier to maintain.

Reply:
I think it's ok to have it like this now and see how things evolve while we develop this endpoint. Long term I expect more paths providing info

Reply:
These metrics are somewhat private. I will break some of them at will. At least some of the filebeat and libbeat queue/output metrics.

Reply:
@urso We didn't realize that in @elastic/apm-server - we were planning on using these in the apm-server monitoring tab - thoughts on how to deal with that?

Reply:
@graphaelli Discussed this recently with @ycombinator and we need to have it more stable form moving forward or at least detect when we break it.